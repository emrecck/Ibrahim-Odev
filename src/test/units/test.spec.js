import Menu from '../../components/Menu.vue'
import Bill from '../../components/Bill.vue'
import SignIn from '../../components/SignIn.vue'
import Navbar from '../../components/Navbar.vue'
import { mount } from '@vue/test-utils'

test('mount Navbar component ', () => {
    const wrapper = mount(Navbar)
    expect(wrapper.exists()).toBe(true)
})

test('mount Menu component ', () => {
    const wrapper = mount(Menu)
    expect(wrapper.html).toBe('<div><p>Foo</p></div>')
})

test('mount Bill component', () => {
    const wrapper = mount(Bill)
    expect(wrapper.name()).toBe('Bill')
})

test('test classes', () => {
    const wrapper = mount(SignIn)
    expect(wrapper.classes('email')).toBe(true)
})

test('add data to Menu', () => {
    const wrapper = mount(Menu)
    const items = wrapper.vm.list
    wrapper.setData({list: [...items, {
        id: 13,
        name: 'ibonun yeri',
        show: false,
        active: false
    }]})
    expect(wrapper.html()).toMatchSnapshot()
})

test('contains test', () => {
    const wrapper = mount(Bill)
    expect(wrapper.contains('div')).toBe(true)
})

test('find test', ()=> {
    const wrapper = mount(Navbar)
    const div = wrapper.find('div')
    expect(div.exists()).toBe(true)
})

test('findAll test', () => {
    const wrapper = mount(Menu)
    const li = wrapper.findAll('li').at(0)

    expect(li.is('div')).toBe(true)
})

test('setProps test', async () => {
    const wrapper = mount(Menu)
    await wrapper.setProps({ ibo: 'ibo' })

    expect(wrapper.vm.ibo).toBe('ibo')
})
