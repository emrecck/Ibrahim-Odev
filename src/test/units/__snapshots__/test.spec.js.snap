// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add data to Menu 1`] = `
<div class="tum-urunler-main col-sm-12 px-0">
  <div class="col-sm-12">
    <ul id="side_category_menu" class="side_category_menu col-sm-12">
      <li id="0" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Aile Boyu </span></router-link>
        <!---->
      </li>
      <li id="1" class="list-item">
        <router-link to="/allproducts"><span class="item-name active">Tüm Ürünler</span></router-link>
        <!---->
      </li>
      <li id="2" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Sebze &amp; Meyve </span></router-link>
        <!---->
      </li>
      <li id="3" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Süt &amp; Peynir </span></router-link>
        <!---->
      </li>
      <li id="4" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Yumurta &amp; Sucuk </span></router-link>
        <!---->
      </li>
      <li id="5" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Salça &amp; Turşu </span></router-link>
        <!---->
      </li>
      <li id="6" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Zeytin &amp; Zeytinyağı </span></router-link>
        <!---->
      </li>
      <li id="7" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Reçel &amp; Bal </span></router-link>
        <!---->
      </li>
      <li id="8" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Sebzeler </span></router-link>
        <!---->
      </li>
      <li id="9" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Ekmek </span></router-link>
        <!---->
      </li>
      <li id="10" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Meyveler </span></router-link>
        <!---->
      </li>
      <li id="11" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Hediye Paketleri </span></router-link>
        <!---->
      </li>
      <li id="12" class="list-item">
        <router-link to="/allproducts"><span class="item-name"> Paketler </span></router-link>
        <!---->
      </li>
    </ul>
  </div>
</div>
`;
